src/webgpu/shader/execution/expression/expression.ts

```diff
@@ -101,7 +101,41 @@ function valueStride(ty: Type): number {
       // vec3s have padding to make them the same size as vec4s
       return 32;
     }
-    unreachable('Matrices of AbstractFloats have not yet been implemented');
+    if (ty instanceof MatrixType) {
+      switch (ty.cols) {
+        case 2:
+          switch (ty.rows) {
+            case 2:
+              return 32;
+            case 3:
+              return 64;
+            case 4:
+              return 64;
+          }
+          break;
+        case 3:
```

Comment:
Since AF matrix is stored as `array<array<AF, rows}>, cols>` and `cols` are not padded in case of 3 cols, should the stride for AF matrix 3xR be as follow?
```ts
        case 3:
          switch (ty.rows) {
            case 2:
              return 48;
            case 3:
              return 96;
            case 4:
              return 96;
          }
          break;
```

Reply:
Thanks for catching this, you are correct.